t/TestCore.pm

```diff
@@ -48,4 +51,47 @@ add_block_preprocessor(sub {
     }
 });
 
+# borrowed from Test::Nginx::Util
+sub gen_random_port (@) {
+    my (@unavailable_port_list) = @_;
+
+    push @unavailable_port_list, $ENV{TEST_NGINX_SERVER_PORT};
+
+    my %unavailable_ports;
+    for my $port (@unavailable_port_list) {
+        $unavailable_ports{$port} = 1;
+    }
+
+    my $server_addr = server_addr;
+
+    srand $$;
+
+    my $random_port;
+    my $tries = 1000;
+    for (my $i = 0; $i < $tries; $i++) {
+        my $port = int(rand 60000) + 1025;
+
+        next if $unavailable_ports{$port};
+
+        my $sock = IO::Socket::INET->new(
+            LocalAddr => $server_addr,
+            LocalPort => $port,
+            Proto => 'tcp',
+            Timeout => 0.1,
+        );
+
+        if (defined $sock) {
+            $sock->close();
+            $random_port = $port;
+            last;
+        }
+    }
+
+    if (!defined $random_port) {
+        bail_out "Cannot find an available listening port number after $tries attempts.\n";
```

Comment:
Always add the `$name - ` prefix the error message otherwise it's hard to debug.